prokube-ai/kubeflow-examples
# Feature/notebook with pytorch training and tensorboard

This PR adds a PyTorch training example using a Variational Autoencoder (VAE) for generating MNIST images. The main objectives are:
1. Demonstrating the feasibility of using Kubeflow notebooks as python IDEs for development.
2. Illustrating the use of TensorBoard within Kubeflow by creating a new instance linked to the PVC of the training notebook.

**Testing Procedure:**
1. Initiate a new notebook instance (preferably with the code server Python image).
2. Clone this repository.
3. Go to the `kf-notebook/mnist` directory.
4. Follow the setup instructions in the README. (Python 3.9 is required for the `poetry install`, which can be installed in a  Conda env in the Kubeflow notebook.)
5. Execute the steps and confirm operational functionality.
